packages/base/src/widget.ts

```diff
@@ -138,6 +140,12 @@ export class JupyterCadPanel extends SplitPanel {
     this._view.delete('axes');
   }
 
+  handleToggleWireframe = () => {
+    if (this.mainViewRef.current) {
```

Comment:
When testing your branch I get a `TypeError: this.mainViewRef is undefined` at runtime here.

How is this `mainViewRef` supposed to be set?

Reply:
`mainViewRef` is supposed to be the reference to the `MainView` component & set like `private mainViewRef: React.RefObject<MainView>;`. However, I don't get this `TypeError: this.mainViewRef is undefined` but `TypeError: Cannot read properties of undefined (reading 'current')` when I try to toggle wireframe. Sorry about this, I'm on my way to look into this and confirm once it gets fixed.

Reply:
the ref object should not be used outside of the object referenced. And you don't need to use the ref here. 

Just follow the approach we are using for toggling the axes or exploded view by updating the `this._view` object. This object is passed to `MainView` object and we listen for the changed signal at https://github.com/jupytercad/JupyterCAD/blob/ca98c7bde6aa5e29592bc4d6152e345400f5f68c/packages/base/src/3dview/mainview.tsx#L1095

Reply:
Thanks for the suggestion @trungleduc, I think my approach was quite novice while i was trying to make it work & I should've discussed more. 
I think this PR needs some rework which I'll be doing after this and would be following what you've suggested :)

Reply:
> Just follow the approach we are using for toggling the axes or exploded view by updating the `this._view` object. This object is passed to `MainView` object and we listen for the changed signal at

Done :)
Thanks again.